### PR TITLE
[BugFix] fix cloudnative table replay bug

### PR DESCRIPTION
**Pull request by kevincai**
* postpone the tablet backend id retrievement until necessary

## What type of PR is this:
- [X] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto backported to target branch
  - [X] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 20 / 25 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|6|9|66.67%|[462, 473, 483]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|4|75.00%|[90]|
|:large_blue_circle:|com/starrocks/load/InsertOverwriteJobRunner.java|5|6|83.33%|[255]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 20 / 25 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|6|9|66.67%|[462, 473, 483]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|4|75.00%|[90]|
|:large_blue_circle:|com/starrocks/load/InsertOverwriteJobRunner.java|5|6|83.33%|[255]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|6|6|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java [hunk 1]
```diff
@@ -245,19 +243,16 @@ private void gc(boolean isReplay) {
             }
             Preconditions.checkState(table instanceof OlapTable);
             OlapTable targetTable = (OlapTable) table;
-            Map<Long, Set<Long>> sourceTabletIds = Maps.newHashMap();
+            Set<Tablet> sourceTablets = Sets.newHashSet();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java:246]: should implement `hashCode` and `equals` methods in `LakeTablet`

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java:246]: ok

**Review by starrocks-xupeng — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java (hunk 1 above)
Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java:246]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 24 / 32 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|7|11|63.64%|[90, 133, 139, 140]|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|6|9|66.67%|[462, 473, 483]|
|:large_blue_circle:|com/starrocks/load/InsertOverwriteJobRunner.java|5|6|83.33%|[255]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|6|6|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Review by zaorangyang — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**